packages/stellar-axelar-std-derive/src/storage.rs

```diff
@@ -21,24 +137,223 @@ enum StorageType {
     Temporary,
 }
 
+impl TryFrom<&[Attribute]> for StorageType {
+    type Error = String;
+
+    fn try_from(attrs: &[Attribute]) -> Result<Self, Self::Error> {
+        attrs
+            .iter()
+            .flat_map(|attr| match attr {
+                _ if attr.path().is_ident("instance") => Some(Self::Instance),
+                _ if attr.path().is_ident("persistent") => Some(Self::Persistent),
+                _ if attr.path().is_ident("temporary") => Some(Self::Temporary),
+                _ => None,
+            })
+            .exactly_one()
+            .map_err(|_| {
+                "storage type must be specified exactly once as 'instance', 'persistent', or 'temporary'"
+                    .to_string()
+            })
+    }
+}
+
 impl StorageType {
-    pub fn storage_method(&self) -> TokenStream {
+    pub fn status_fns(
+        &self,
+        getter_name: &Ident,
+        setter_name: &Ident,
+        remover_name: &Ident,
+        param_list: &TokenStream,
+        key: &TokenStream,
+    ) -> TokenStream {
         match self {
-            Self::Instance => quote! { instance },
-            Self::Persistent => quote! { persistent },
-            Self::Temporary => quote! { temporary },
+            Self::Persistent => quote! {
+                pub fn #getter_name(#param_list) -> bool {
+                    let value = env.storage()
+                        .persistent()
+                        .has(&#key);
+
+                    if value {
+                        stellar_axelar_std::ttl::extend_persistent_ttl(env, &#key)
+                    }
+
+                    value
+                }
+
+                pub fn #setter_name(#param_list) {
+                    env.storage()
+                        .persistent()
+                        .set(&#key, &());
+                }
+
+                pub fn #remover_name(#param_list) {
```

Comment:
This duplication seems unnecessary? Have a separate match block first that creates the ttl method token stream and the storage type name. Then, all methods can be defined generically?